src/variational/VariationalInference.jl

```diff
@@ -1,50 +1,329 @@
+        kwargs...
+    )
+
+Find a numerically non-degenerate Gaussian `q` with a scale with full-rank factors (traditionally referred to as a "full-rank family") for approximating the target `model`.
+
+# Arguments
+- `model`: The target `DynamicPPL.Model`.
+
+# Keyword Arguments
+- `location`: The location parameter of the initialization. If `nothing`, a vector of zeros is used.
+- `scale`: The scale parameter of the initialization. If `nothing`, an identity matrix is used.
+
+The remaining keyword arguments are passed to `q_locationscale`.
+
+# Returns 
+- `q::Bijectors.TransformedDistribution`: A `AdvancedVI.LocationScale` distribution matching the support of `model`.
+"""
+function q_fullrank_gaussian(
+    rng::Random.AbstractRNG,
+    model::DynamicPPL.Model;
+    location::Union{Nothing,<:AbstractVector}=nothing,
+    scale::Union{Nothing,<:LowerTriangular}=nothing,
+    kwargs...,
+)
+    return q_locationscale(
+        rng, model; location, scale, meanfield=false, basedist=Normal(), kwargs...
+    )
+end
+
+function q_fullrank_gaussian(model::DynamicPPL.Model; kwargs...)
+    return q_fullrank_gaussian(Random.default_rng(), model; kwargs...)
+end
+
+"""
+    vi(
+        [rng::Random.AbstractRNG,]
+        model::DynamicPPL.Model;
+        q,
+        n_iterations::Int;
+        objective::AdvancedVI.AbstractVariationalObjective = AdvancedVI.RepGradELBO(
+            10; entropy = AdvancedVI.ClosedFormEntropyZeroGradient()
+        ),
+        show_progress::Bool = Turing.PROGRESS[],
+        optimizer::Optimisers.AbstractRule = AdvancedVI.DoWG(),
+        averager::AdvancedVI.AbstractAverager = AdvancedVI.PolynomialAveraging(),
+        operator::AdvancedVI.AbstractOperator = AdvancedVI.ProximalLocationScaleEntropy(),
+        adtype::ADTypes.AbstractADType = Turing.DEFAULT_ADTYPE,
+        kwargs...
+    )
+
+Approximating the target `model` via variational inference by optimizing `objective` with the initialization `q`.
+This is a thin wrapper around `AdvancedVI.optimize`.
+
+# Arguments
+- `model`: The target `DynamicPPL.Model`.
+- `q`: The initial variational approximation.
+- `n_iterations`: Number of optimization steps.
+
+# Keyword Arguments
+- `objective`: Variational objective to be optimized.
+- `show_progress`: Whether to show the progress bar.
+- `optimizer`: Optimization algorithm.
+- `averager`: Parameter averaging strategy.
+- `operator`: Operator applied after each optimization step.
+- `adtype`: Automatic differentiation backend.
+
+See the docs of `AvancedVI.optimize` for additional keyword arguments.
+
+# Returns 
+- `q`: Variational distribution formed by the last iterate of the optimization run.
+- `q_avg`: Variational distribution formed by the averaged iterates according to `averager`.
+- `state`: Collection of states used for optimization. This can be used to resume from a past call to `vi`.
+- `info`: Information generated during the optimization run.
+"""
+function vi(
```

Comment:
This function is a thin wrapper around `AdvancedVI.optimize`. I'd suggest we consider https://github.com/TuringLang/Turing.jl/issues/2509#issue-2923182002 for this interface to make it future-proof: 

```
optimise(model, VI(q, n_iterations, objective; ...), ...)
```

Reply:
Given that the function `vi` itself is more of a legacy from the v0.2 days, I suggest we first go with this and add the new interface in the future? Whatever we decide to do, I think having `vi` now won't be too much of a hassle in the future. (On that matter, I am sympathetic to the proposal to unify everything into `infer`)